src/component/2d/zones/Signal.jsx

```diff
@@ -0,0 +1,72 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import { memo, useState, useEffect } from 'react';
+
+import { useAssignment } from '../../assignment';
+import { useChartData } from '../../context/ChartContext';
+import { useHighlightData, useHighlight } from '../../highlight';
+import { HighlightSignalConcatenation } from '../../panels/extra/constants/ConcatenationStrings';
+import { get2DXScale, get2DYScale } from '../utilities/scale';
+
+const Signal = memo(({ signal, signalID }) => {
+  const { margin, width, height, xDomain, yDomain } = useChartData();
+  const scaleX = get2DXScale({ margin, width, xDomain });
+  const scaleY = get2DYScale({ margin, height, yDomain });
+
+  const x = scaleX(signal.x.delta);
+  const y = scaleY(signal.y.delta);
+
+  const assignment = useAssignment(signalID);
+  const highlight = useHighlight([
+    assignment.id,
+    `${assignment.id}${HighlightSignalConcatenation}X`,
+    `${assignment.id}${HighlightSignalConcatenation}Y`,
```

Comment:
I saw that you use this combination of ID frequently in different places can create help function that you pass assignment object and return the ID.